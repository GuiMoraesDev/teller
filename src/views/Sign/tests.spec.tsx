import React from 'react';

import { render } from 'tests/test.utils';

import SignTemplate from './template';

const ContainerRender = (mockSubmit: jest.Mock<any, any>) =>
	render(
		<SignTemplate
			type="signin"
			title="Welcome to Teller"
			description="Use one social link above to sign in"
			handleLoginSuccess={mockSubmit}
			buttonProps={{
				label: 'Jest test',
			}}
		/>
	);

describe('Login page', () => {
	it('should render correctly', () => {
		const mockSubmit = jest.fn();

		const { container } = ContainerRender(mockSubmit);

		expect(container.firstChild).toMatchSnapshot();
	});
});
