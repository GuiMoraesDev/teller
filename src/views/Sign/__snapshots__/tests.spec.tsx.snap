// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page should render correctly 1`] = `
.c5 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  height: 4.4rem;
  border: 0.1rem solid #C9CAC8;
  border-radius: 0.4rem;
  margin: 0;
  padding: 0 1.4rem;
  gap: 1.2rem;
  text-transform: capitalize;
  -webkit-transition: background-color 0.2s, color 0.2s;
  transition: background-color 0.2s, color 0.2s;
  cursor: pointer;
  color: #FDFDFD;
  background-color: #63989F;
  border-color: #4F7A7F;
}

.c5:hover {
  background-color: #82ADB2;
}

.c5 svg {
  color: inherit;
}

.c5:hover {
  background-color: #4F7A7F;
}

.c5 .spin {
  -webkit-animation-name: ieAXeN;
  animation-name: ieAXeN;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
}

.c8 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  height: 4.4rem;
  border: 0.1rem solid #C9CAC8;
  border-radius: 0.4rem;
  margin: 0;
  padding: 0 1.4rem;
  gap: 1.2rem;
  text-transform: none;
  -webkit-transition: background-color 0.2s, color 0.2s;
  transition: background-color 0.2s, color 0.2s;
  cursor: pointer;
  color: #181917;
  background-color: transparent;
  border-color: transparent;
}

.c8:hover {
  background-color: #82ADB2;
}

.c8 svg {
  color: inherit;
}

.c8:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #181917;
  background-color: transparent;
}

.c8 .spin {
  -webkit-animation-name: ieAXeN;
  animation-name: ieAXeN;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
}

.c1 {
  position: relative;
  padding: 0;
  margin: 0;
  font-size: 4.8rem;
  font-weight: 500;
  font-family: 'Montserrat',sans-serif;
  line-height: 8.0rem;
  color: #181917;
  text-transform: none;
  box-sizing: border-box;
}

.c2 {
  position: relative;
  padding: 0;
  margin: 0;
  font-size: 2.4rem;
  font-weight: 500;
  font-family: 'Montserrat',sans-serif;
  line-height: 4.0rem;
  color: #181917;
  text-transform: capitalize;
  box-sizing: border-box;
}

.c4 {
  position: relative;
  padding: 0;
  margin: 0;
  font-size: 1.6rem;
  font-weight: 400;
  font-family: 'Roboto',sans-serif;
  line-height: 2.4rem;
  color: #181917;
  text-transform: none;
  box-sizing: border-box;
}

.c7 {
  position: relative;
  padding: 0;
  margin: 0;
  font-size: 1.4rem;
  font-weight: 400;
  font-family: 'Roboto',sans-serif;
  line-height: 2.0rem;
  color: #181917;
  text-transform: none;
  box-sizing: border-box;
}

.c0 {
  grid-area: Content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 2.0rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  max-width: 75%;
  border-radius: 0.4rem;
  gap: 2.0rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 50%;
  gap: 1.2rem;
  margin: 1.2rem 0;
  padding: 1.2rem 0;
  border-top: 0.1rem solid #C9CAC8;
  border-bottom: 0.1rem solid #C9CAC8;
}

<main
  class="c0"
>
  <strong
    class="c1"
  >
    Welcome to Teller
  </strong>
  <h2
    class="c2"
  >
    sign in
  </h2>
  <form
    class="c3"
    id="sign-form"
  >
    <p
      class="c4"
    >
      Use the form bellow to sign in
    </p>
  </form>
  <button
    class="c5"
    form="sign-form"
    type="submit"
  >
    sign in
  </button>
  <div
    class="c6"
  >
    <p
      class="c7"
    >
      Or use social links: 
    </p>
    <div
      id="google-sign"
    />
  </div>
  <button
    class="c8"
  >
    Jest test
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        fill="none"
        height="256"
        width="256"
      />
      <path
        d="M122.3,71.4l19.8-19.8a44.1,44.1,0,0,1,62.3,62.3l-28.3,28.2a43.9,43.9,0,0,1-62.2,0"
        fill="none"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="16"
      />
      <path
        d="M133.7,184.6l-19.8,19.8a44.1,44.1,0,0,1-62.3-62.3l28.3-28.2a43.9,43.9,0,0,1,62.2,0"
        fill="none"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="16"
      />
    </svg>
  </button>
</main>
`;
